splatnet2statink.py

```diff
@@ -273,37 +328,18 @@ def monitor_battles(s_flag, t_flag, r_flag, secs, debug):
 	except:
 		pass
 
-	battles = [] # 50 recent battles on splatnet
+	if r_flag:
+		battles = upload_recent(s_flag, t_flag, debug)
+	else:
+		battles = [] # 50 recent battles on splatnet
 	wins = 0
 	losses = 0
 
-	# if r_flag, check if there are any battles in splatnet that aren't on stat.ink
-	if r_flag:
-		print "Checking if there are previously-unuploaded battles..."
-		printed = False
-		url  = 'https://stat.ink/api/v2/user-battle?only=splatnet_number&count=50'
-		auth = {'Authorization': 'Bearer ' + API_KEY}
-		resp = requests.get(url, headers=auth)
-		statink_battles = json.loads(resp.text) # 50 recent battles on stat.ink
-
-	for result in results[::-1]:
```

Comment:
In python, does this mean "iterate through these in reverse"?

Reply:
Yes, the `-1` means to go through the array in reverse order.

Reply:
- `[::-1]` makes a reversed copy of a list (not really necessary here, i guess…)
- `reversed()` returns an iterator and walks through a list in reverse order.
- and just for good measure `list.reverse()` is in-place list reversal.

Reply:
Thanks! I’m most comfortable in ruby so python isn’t far off, but syntax like this isn’t as easy to search for.

Reply:
i 100% love ruby but i don't know ruby

MINASWAN